modules/pam_env/pam_env.c

```diff
@@ -568,14 +575,107 @@ _pam_get_item_byname(pam_handle_t *pamh, const char *name)
   return itemval;
 }
 
+static void
+_strbuf_init(struct string_buffer *buffer)
+{
+  buffer->len = 0;
+  buffer->max = 0;
+  buffer->str = NULL;
+}
+
+static void
+_strbuf_free(struct string_buffer *buffer)
+{
+  pam_overwrite_n(buffer->str, buffer->len);
+  _pam_drop(buffer->str);
+  buffer->len = 0;
+  buffer->max = 0;
+}
+
+/*
+ * Allocates given amount of bytes in buffer for addition.
+ * Internally adding an extra byte for NUL character.
+ *
+ * Returns 0 on success, 1 on error.
+ */
+static int
+_strbuf_reserve(struct string_buffer *buffer, size_t add)
+{
+  char *p;
+  size_t s;
+
+  if (buffer->max >= SIZE_MAX - add) {
+    return 1;
+  }
+
+  if (buffer->len + add < buffer->max) {
+    return 0;
+  }
```

Comment:
Heavily dependent on personal taste. Does the required change fix a bug?

Reply:
No, but makes two things more obvious:

1. That the main assumption is `buffer->len < buffer->max`
2. That the check is to test for remaining space

Nothing that fixes a bug, but makes the code a bit more obvious to read.

Reply:
Moved up for technical advantage: The majority of cases will be that enough space is already available, since char-by-char addition takes place. Leaving the function as soon as possible saves CPU cycles.

Reply:
Otherwise I'm not interested in favorite bike shed colors. Neither now nor in the future.

If my proposed patch violates existing code style or practices within the project, I'm more than willing to adjust it.

Reply:
Speaking about it, I've adjusted `max` to `size` to comply with the buffer mechanism available in pam_handlers.c. Seems it was okay there.